src/main/java/org/jabref/gui/integrity/IntegrityCheckDialog.java

```diff
@@ -116,4 +171,73 @@ public void clearFilters() {
             });
         }
     }
+
+    private void removeRowFromTable(IntegrityMessage message) {
+        ObservableList<IntegrityMessage> mutableMessages = FXCollections.observableArrayList(messagesTable.getItems());
```

Comment:
You should delete the item in the viewModel - you would bind the table via messagesTablel.itemsProperty().bind(viewModel.messagesListProperty() or someting similar
see e..g  https://github.com/JabRef/jabref/blob/7c958e7dd067adbfd7244224aa551c6b3f1e9721/src/main/java/org/jabref/gui/preferences/table/TableTab.java#L98